File: 1299. Replace Elements with Greatest Element on Right Side.py
class Solution(object):
    def replaceElements(self, arr):
        res = [None] * len(arr)
        res[-1] = -1
        for i in range(len(arr) - 1, 0, -1):
            res[i - 1] = max(arr[i], res[i])
        return res
        

File: 1299. Replace Elements with Greatest Element on Right Side.rs
impl Solution {
    pub fn replace_elements(arr: Vec<i32>) -> Vec<i32> {
        let mut res = vec![0; arr.len()];
        res[arr.len() - 1] = -1;
        for i in (1..arr.len()).rev() {
            res[i - 1] = arr[i].max(res[i]);
        }
        res
    }
}
